packages/colorpickers/src/styled/ColorSwatch/StyledCheckIcon.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import { parseToRgb, readableColor } from 'polished';
+import styled, { ThemeProps, DefaultTheme } from 'styled-components';
+import { retrieveComponentStyles, DEFAULT_THEME } from '@zendeskgarden/react-theming';
+import CheckIcon from '@zendeskgarden/svg-icons/src/12/check-sm-fill.svg';
+import { IRGBColor } from '../../utils/types';
+
+const COMPONENT_ID = 'colorpickers.colorswatch_check';
+
+interface IStyledCheckIcon {
+  color: string;
+}
+
+const colorStyles = (props: IStyledCheckIcon & ThemeProps<DefaultTheme>) => {
+  const { theme, color } = props;
+  const { alpha } = parseToRgb(color) as IRGBColor;
+  let checkColor = readableColor(color, theme.colors.foreground, theme.colors.background);
+
+  if (alpha !== undefined && alpha < 0.4) {
+    checkColor = theme.colors.foreground;
+  }
+
+  return `
+    color: ${checkColor}
+  `;
+};
+
+export const StyledCheckIcon = styled(CheckIcon)`
```

Comment:
[nit] `color` (and `theme`) props should not be getting spread as attributes on to the rendered SVG.